jgrapht-core/src/main/java/org/jgrapht/generate/GraphGenerator.java

```diff
@@ -49,9 +85,22 @@
      * @param vertexFactory called to produce new vertices
      * @param resultMap if non-null, receives implementation-specific mappings from String roles to
      *        graph elements (or collections of graph elements)
+     * @deprecated Use simpler methods
      */
-    void generateGraph(
-        Graph<V, E> target, VertexFactory<V> vertexFactory, Map<String, T> resultMap);
+    @Deprecated
+    default void generateGraph(
+        Graph<V, E> target, VertexFactory<V> vertexFactory, Map<String, T> resultMap)
+    {
+        boolean wasNull = false;
+        if (vertexFactory != null && target.getVertexSupplier() == null) {
+            target.setVertexSupplier(vertexFactory::createVertex);
+            wasNull = true;
+        }
+        generateGraph(target, resultMap);
```

Comment:
This should use a try/finally block to make sure the vertex supplier isn't permanently altered.  But better would be to avoid modifying the supplier at all.  I think there may be an easy way to do this using GraphDelegator, overriding only the addVertex() method.

Reply:
Not needed anymore, since the change happens on a temporary delegator.